test/unit/specs/components/Audiences/index.spec.js

```diff
@@ -0,0 +1,46 @@
+/*
```

Comment:
Generally, the problem with writing unit tests for the index.js files is that its impossible to inject mocks for dependencies. The index files do the injecting. In this case you aren't able to mock the processDestinations function, and indeed it is getting imported and called. The index.js files should just be wiring up all the dependencies together and shouldn't be doing any work.  I recommend creating a "injectProcessResponse.js" file like this:

```
export default ({ processDestinations }) => {

  const processDestinationsFromResponse = ({ response }) => {
     const destinations = response.getPayloadsByType("activation:push");
     return processDestinations(destinations);
   };

   const retrievePullDestinations = ({ response }) => {
     return Promise.resolve({
       destinations: response.getPayloadsByType("activation:pull")
     });
   };

   return ({ response }) => {
     return processDestinationsFromResponse({ response }).then(() =>
       retrievePullDestinations({ response })
     );
   };
};
```

Then index.js looks like this:

```
import { fireReferrerHideableImage } from "../../utils";
import injectProcessDestinations from "./injectProcessDestinations";
import injectProcessResponse from "./injectProcessResponse";

const createAudiences = ({ logger }) => {
  const processDestinations = injectProcessDestinations({
     fireReferrerHideableImage,
     logger
   });

   const processResponse = injectProcessResponse({ processDestinations });

   return {
     lifecycle: {
       onResponse: processResponse
     },
     commands: {}
   };
};

```

Then you can move this file to injectProcessResponse.spec.js and create a mock for processDestinations. You can ensure that processDestinations is called correctly, and that "activation:pull" handles are returned.

Reply:
That's how you know I'm terribly rusty :) I'm usually obsessed with dependency injection; not anymore! :) :)